### DISCUSSION TITLE
Count the selected chains

### DISCUSSION THREAD

**Discussion by Marwa-Da** [molstar/molstar#959] [Q&A]
Hi,

I'm trying to count the chains selected by the user and detect the name of them where I use the auth_asym_id identifier for detecting the chains (this is opposed to the stats control implementation where the label_asym_id is used).
Any hints on how to start?

**Comment by dsehnal** [comment]
Something along these lines:

```ts
const uniqueChains = new Set<string>();
const s = ...; // struture you want the selection for
const sel = plugin.managers.structure.selection.getStructure(s);
if (sel) {
    Structure.eachAtomicHierarchyElement(sel, {
        residue: e => uniqueChains.add(StructureProperties.chain.auth_asym_id(e));
    });
}
```

``chain:`` instead of ``residue:`` should work also and be a bit faster. Tho each chain there will correspond to the `label_asym_id` chains.
